### PR TITLE
Adds HTTP Tracing to colly requests

### PR DESCRIPTION
**Pull request by sdab**
Go HTTP supports tracing with the "net/http/httptrace" package. This adds a very
basic HTTPTrace object to the HTTP Request that tracks HTTP request timing. **This
is useful for users wanting to tune their crawling based on RTT.**

The resulting trace object is added to the colly Response which is accessible
to a caller via any of the Response callbacks.

Commit includes unit tests for the new HTTPTrace object as well as a new case for colly_test.

### CODE REVIEW

**Comment by sdab** [comment]
Wanted to check in if this is something that would be useful upstream. Happy to change my approach if needed.

**Comment by asciimoo** [comment]
This is a useful addition, thanks. Do you have any stats about how many memory/cpu overhead is generated by enabling this feature?
Perhaps, it would be good to turn it off somehow if the performance matters but no stats necessary in a scraping job.

**Comment by sdab** [comment]
Sorry, for the late response. I missed the notification.

I have made the feature optional and off by default to avoid overhead.

File: response.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ type Response struct {
 	Request *Request
 	// Headers contains the Response's HTTP headers
 	Headers *http.Header
+	// Trace contains the HTTPTrace for the request.
```
[the remark this answers is not in the document]
Comment (sdab) [reply response.go]: Done

**Pull request merged**